fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -65,51 +64,37 @@
     private static final Logger LOG = LoggerFactory.getLogger(MetricUnionAvailability.class);
 
     private final Set<TableName> dataSourceNames;
-    private final Set<MetricColumn> metricColumns;
-    private final Map<Availability, Set<MetricColumn>> availabilitiesToAvailableColumns;
+    private final Set<String> metricNames;
+    private final Map<Availability, Set<String>> availabilitiesToMetricNames;
 
     /**
      * Constructor.
      *
      * @param physicalTables  A set of <tt>PhysicalTable</tt>s whose Dimension schemas are the same and
-     * the Metric Schemas are fully different(i.e. no overlap) on every table
+     * the Metric columns are unique(i.e. no overlap) on every table
      * @param columns  The set of all configured columns, including dimension columns, that metric union availability
      * will respond with
      */
     public MetricUnionAvailability(@NotNull Set<PhysicalTable> physicalTables, @NotNull Set<Column> columns) {
-        metricColumns = Utils.getSubsetByType(columns, MetricColumn.class);
+        metricNames = Utils.getSubsetByType(columns, MetricColumn.class).stream()
+                .map(MetricColumn::getName)
+                .collect(Collectors.toSet());
 
-        // get a map from availability to its available metric columns intersected with configured metric columns
-        // i.e. metricColumns
-        availabilitiesToAvailableColumns = physicalTables.stream()
+        // map each metric to its corresponding availability object
```

Comment:
This comment change is losing the fact that this is intersected with the configured (ie. passed in) metric columns.

This comment change is also losing the fact that this is building `Map<Availability, Set<metric column>>`, rather than `Map<metric column, Availability>`, which seems to be what the comment is saying